### PR TITLE
Fix: [GMTDate] .toJvmDate() uses current date instead of given date.

### PR DESCRIPTION
**Pull request by talanov**
Solves https://github.com/ktorio/ktor/issues/984

### CODE REVIEW

**Comment by e5l** [comment]
Hi @talanov, thanks for the PR.

Could you add a couple of tests with timezones?

**Comment by talanov** [comment]
Hi @e5l, thanks for reviewing.

Added tests.

File: ktor-utils/jvm/test/io/ktor/tests/utils/DateJvmTest.kt [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+        assertEquals(dateRaw.timestamp, date.time)
+    }
+
+    @Test
+    fun testJvmDateWithOffSetDateTime() {
+        val dateRaw = GMTDate(20, 20, 20, 20, Month.FEBRUARY, 20)
+        val date: OffsetDateTime = OffsetDateTime.of(dateRaw.year, dateRaw.month.ordinal, dateRaw.dayOfMonth, dateRaw.hours, dateRaw.minutes, dateRaw.seconds, 0, ZoneOffset.UTC)
+
+        assertEquals(dateRaw.timestamp, date.toEpochSecond())
```
Comment (e5l) [inline ktor-utils/jvm/test/io/ktor/tests/utils/DateJvmTest.kt]: It looks like you try to compare different quantities: 
```
java.lang.AssertionError: expected:<-61531846780000> but was:<-61534352380>
```

Comment (talanov) [reply ktor-utils/jvm/test/io/ktor/tests/utils/DateJvmTest.kt]: Yeah, my bad, sorry.

Fixed, added 1 test.
```
BUILD SUCCESSFUL in 18s
123 actionable tasks: 4 executed, 119 up-to-date
```

**Pull request merged**